src/mixins/localHooks.js

```diff
@@ -34,11 +34,22 @@ const localHooks = {
    * Run hooks.
    *
    * @param {string} key The hook name.
-   * @param {*} params Additional parameters passed to callback function.
+   * @param {*} [arg1] Additional parameter passed to callback function.
+   * @param {*} [arg2] Additional parameter passed to callback function.
+   * @param {*} [arg3] Additional parameter passed to callback function.
+   * @param {*} [arg4] Additional parameter passed to callback function.
+   * @param {*} [arg5] Additional parameter passed to callback function.
+   * @param {*} [arg6] Additional parameter passed to callback function.
    */
-  runLocalHooks(key, ...params) {
+  runLocalHooks(key, arg1, arg2, arg3, arg4, arg5, arg6) {
     if (this._localHooks[key]) {
-      arrayEach(this._localHooks[key], callback => callback.apply(this, params));
+      const length = this._localHooks[key].length;
+
+      // Do not optimize this loop with the arrayEach or arrow function! If you do, You'll
+      // decrease perf because of GC. The "...rest" ES6+ syntax as well decreases the performance.
```

Comment:
```suggestion
      // don't optimize this loop with the `arrayEach()` method or arrow functions
      // otherwise, performance will decrease because of garbage collection
      // using the `...rest` syntax (ES6 and later) will decrease performance as well
```

Reply:
The `arrayEach()` is a function without context (object owner). So, technically it should be named as a function. WDYT to rephrase the sentence to `...`arrayEach()` function or arrow functions...`?

Reply:
@budnix - sure, makes sense!